src/offset.js

```diff
@@ -57,10 +57,10 @@ jQuery.offset = {
 		}
 
 		if ( options.top != null ) {
-			props.top = ( options.top - curOffset.top ) + curTop;
+			props.top = ( options.top - ( curOffset ? curOffset.top : 0 ) ) + curTop;
 		}
 		if ( options.left != null ) {
-			props.left = ( options.left - curOffset.left ) + curLeft;
+			props.left = ( options.left - ( curOffset ? curOffset.left : 0 ) ) + curLeft;
```

Comment:
Did you try leaving these lines alone in favor of adjusting the above to `curOffset = curElem.offset() || { top: 0, left: 0 }`? It strikes me as more straightforward, and possibly smaller.